decryptor/mysql/packet.go

```diff
@@ -128,7 +128,7 @@ func (packet *MysqlPacket) replaceQuery(newQuery string) {
 
 // readPacket read header to struct and return payload as return result or error
 func (packet *MysqlPacket) readPacket(connection net.Conn) ([]byte, error) {
-	if _, err := connection.Read(packet.header); err != nil {
+	if _, err := io.ReadFull(connection, packet.header); err != nil {
```

Comment:
what is the difference?

Reply:
helper read more than once (if first read call was not full and return only part of data cached in os buffer). until some error or until no data was returned. ReadFull read in loop until read expected num of bytes. I meet one failed test when here read return part of packet and without error